tests/test_web_urldispatcher.py

```diff
@@ -458,3 +459,47 @@ async def test_static_absolute_url(aiohttp_client: Any, tmp_path: Any) -> None:
     client = await aiohttp_client(app)
     resp = await client.get("/static/" + str(file_path.resolve()))
     assert resp.status == 403
+
+
+@pytest.mark.xfail(
+    raises=AssertionError,
+    reason="Regression in v3.7: https://github.com/aio-libs/aiohttp/issues/5621",
+)
+async def test_decoded_match_regex(aiohttp_client) -> None:
```

Comment:
Looking at this closer, I think it's actually one test with multiple params so let's just parametrize it:
```suggestion
@pytest.mark.parametrize(
    ("route_definition", "urlencoded_path", "expected_http_resp_status"),
    (
        ("/{user_ids:([0-9]+)(,([0-9]+))*}/hello", "/467%2C802%2C24834/hello", 200),
        ("/1%2C3/hello, "/1%2C3/hello", 404),
    ),
    ids=("urldecoded_route", "urlencoded_route"),
)
async def test_decoded_url_match(
    aiohttp_client,
    route_definition,
    urlencoded_path,
    expected_http_resp_status,
) -> None:
```

Reply:
Or maybe even
```suggestion
@pytest.mark.parametrize(
    ("route_definition", "urlencoded_path", "expected_http_resp_status"),
    (
        ("/467,802,24834/hello", "/467%2C802%2C24834/hello", 200),
        ("/1%2C3/hello, "/1%2C3/hello", 404),
    ),
    ids=("urldecoded_route", "urlencoded_route"),
)
async def test_decoded_url_match(
    aiohttp_client,
    route_definition,
    urlencoded_path,
    expected_http_resp_status,
) -> None:
```

Reply:
Looking at this closer, I think it's actually one test with multiple params so let's just parametrize it:
```suggestion
@pytest.mark.parametrize(
    ("route_definition", "urlencoded_path", "expected_http_resp_status"),
    (
        ("/467,802,24834/hello", "/467%2C802%2C24834/hello", 200),
        ("/{user_ids:([0-9]+)(,([0-9]+))*}/hello", "/467%2C802%2C24834/hello", 200),
        ("/1%2C3/hello, "/1%2C3/hello", 404),
    ),
    ids=("urldecoded_route", "urldecoded_route_with_regex", "urlencoded_route"),
)
async def test_decoded_url_match(
    aiohttp_client,
    route_definition,
    urlencoded_path,
    expected_http_resp_status,
) -> None:
```

Reply:
@thehesiod with this suggestion, I believe you can just accept the patches via GH UI and then pull+tweak the other test removal locally.